pep-0710.rst

```diff
@@ -0,0 +1,357 @@
+--------------------------------
+
+:pep:`610` and `its corresponding canonical PyPA spec
+<https://packaging.python.org/en/latest/specifications/direct-url/>`__
+recommend stating the ``hashes`` field of the ``archive_info`` in the
+``direct_url.json`` file but allows ignoring it under certain circumstances
+following :rfc:`2119`:
+
+  A hashes key SHOULD be present as a dictionary mapping a hash name to a hex
+  encoded digest of the file.
+
+This PEP enforces availability of the ``hashes`` field of the ``archive_info``
+in the ``provenance_url.json`` file if ``provenance_url.json`` file is created:
+
+  The value of ``archive_info`` MUST be a dictionary with a single key
+  ``hashes``.
+
+By doing so, consumers of ``provenance_url.json`` file can perform check on
+artifact digests when ``provenance_url.json`` file is created by installers.
+
+Open Issues
+===========
+
+Availability of the provenance_url.json file in Conda
+-----------------------------------------------------
+
+We would like to get feedback on the ``provenance_url.json`` file by Conda
+maintainers or developers. It is not clear whether Conda would like to adopt
+the ``provenance_url.json`` file.
+
+Using provenance_url.json in downstream installers
+--------------------------------------------------
+
+The proposed ``provenance_url.json`` file was meant to be adopted primarily by
+Python installers. Other installers, such as apt or dnf, might record
+provenance of the installed downstream Python distributions in their specific
+way that can be specific to downstream package management. The proposed file is
+not expected to be created by these downstream package installers and thus they
+were intentionally left out of this PEP. However, any input by developers or
+maintainers of these installers is valuable to possibly enrich the
+``provenance_url.json`` file with information that would help in some way.
+
+Backwards Compatibility
+=======================
+
+Since this PEP specifies a new file in the ``*.dist-info`` directory, there are
+no backwards compatibility implications to consider in the ``provenance_url.json``
+file itself. Also, this proposal does not make any changes to the
+``direct_url.json`` described in :pep:`610` and `its corresponding canonical
+PyPA spec
+<https://packaging.python.org/en/latest/specifications/direct-url/>`__.
+
+The content of ``provenance_url.json`` file was designed in a way to eventually
+allow installers reuse some of the logic supporting :pep:`610` when a
+direct URL refers to a source archive or a wheel.
+
+References
+==========
+
+The following changes were done to pip to support this PEP:
+
+.. [1] `A patch to pip introducing provenance_url.json as discussed in this PEP
+  <https://github.com/fridex/pip/pull/1/>`__
+
+.. [2] `Changes to pip to support the decision for creating
+  provenance_url.json instead of stating provenance in already existing
+  direct_url.json <https://github.com/fridex/pip/pull/2/>`__
```

Comment:
```suggestion
.. _fridex/pip#1: https://github.com/fridex/pip/pull/1/

.. _fridex/pip#2: https://github.com/fridex/pip/pull/2/
```

* [Per PEP 12](https://peps.python.org/pep-0012/#hyperlinks), use normal direct links with an explicit name, rather than requiring the reader to click a tiny footnote, and then click the actual link they want.
* Use a more descriptive title for the reference, helping readers in general and a11y in particular
* Avoid hardcoded numbered footnotes, as they are fragile and very easy to break once editing the PEP (plus, they're out of order here).